benches/throughput.rs

```diff
@@ -374,12 +380,138 @@ pub fn metrics(c: &mut Criterion) {
+    }
+}
+
+/// Reserve IPs in 127.0.1.0/24 for these HBONE connection tests.
+/// Thus, we identify hosts by a u8 which represents an IP in the form 127.0.1.x.
+fn hbone_connection_ip(x: u8) -> IpAddr {
+    IpAddr::V4(Ipv4Addr::new(127, 0, 1, x))
+}
+
+fn hbone_connection_config() -> ztunnel::config::ConfigSource {
+    let mut workloads: Vec<LocalWorkload> = Vec::with_capacity(MAX_HBONE_WORKLOADS as usize);
+    // We can't create one work load with many IPs because ztunnel could connect to any one causing
+    // inconsistent behavior. Instead, we create one workload per IP.
+    for i in 1..MAX_HBONE_WORKLOADS + 1 {
+        let lwl = LocalWorkload {
+            workload: Workload {
+                workload_ips: vec![hbone_connection_ip(i)],
+                protocol: Protocol::HBONE,
+                uid: format!("cluster1//v1/Pod/default/local-source{}", i),
+                name: format!("workload-{}", i),
+                namespace: format!("namespace-{}", i),
+                service_account: format!("service-account-{}", i),
+                ..test_helpers::test_default_workload()
+            },
+            services: Default::default(),
+        };
+        workloads.push(lwl);
+    }
+
+    let lc = ztunnel::xds::LocalConfig {
+        workloads,
+        policies: vec![],
+        services: vec![],
+    };
+    let mut b = bytes::BytesMut::new().writer();
+    serde_yaml::to_writer(&mut b, &lc).ok();
+    let b = b.into_inner().freeze();
+    ztunnel::config::ConfigSource::Static(b)
+}
+
+/// Benchmark how long it takes to establish a new HBONE connection.
+/// This is tricky because ztunnel will keep a connection pool.
+/// Repeated connections from the same source to the same destination will use the pooled
+/// connection. Instead, we register MAX_HBONE_WORKLOADS giving us O(MAX_HBONE_WORKLOADS^2)
+/// source/destination IP combinations which is (hopefully) enough.
+fn hbone_connections(c: &mut Criterion) {
+    let rt = tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .unwrap();
+
+    // Global setup: spin up an echo server and ztunnel instance
+    let (echo_addr, ta) = rt.block_on(async move {
+        let cert_manager = identity::mock::new_secret_manager(Duration::from_secs(10));
+        let port = 80;
+        let config_source = Some(hbone_connection_config());
+        let config = test_helpers::test_config_with_port_xds_addr_and_root_cert(
+            port,
+            None,
+            None,
+            config_source,
+        );
+        let app = app::build_with_cert(config, cert_manager.clone())
+            .await
+            .unwrap();
+        let ta = TestApp::from((&app, cert_manager));
+        ta.ready().await;
+
+        let echo = tcp::TestServer::new(Mode::ReadWrite, 0).await;
+        let echo_addr = echo.address();
+        let _ = tokio::spawn(async move {
+            let _ = tokio::join!(app.wait_termination(), echo.run());
+        });
```

Comment:
Adding a `.await` here should fix the CI

Reply:
TODO: Drop works, but hoping that someone has a better solution than just orphaning the tasks.